src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/HeaderValidationProcessor.java

```diff
@@ -41,7 +45,21 @@ public void process(Exchange exchange) throws Exception {
         RouteProperties routeProperties = (RouteProperties) exchange.getIn().getHeader(MappingConstants.ROUTE_DETAILS);
         String csv = exchange.getIn().getBody(String.class);
         CSVReader reader = new CSVReader(new StringReader(csv));
-        String[] header = reader.readNext();
+        String[] actualCsvHeaders = reader.readNext();
+        String isHeaderValidationEnabled = routeProperties.getIsHeaderValidationEnabled();
+        String expectedCsvHeaders = routeProperties.getCsvHeadersExpected();
+
+        Predicate<String> checkIfHeaderValidationEnabled = (isEnabled) -> isEnabled != null
+                && isEnabled.equalsIgnoreCase(Boolean.TRUE.toString());
+        boolean isEnabled = checkIfHeaderValidationEnabled.test(isHeaderValidationEnabled);
+
+        if (isEnabled && isNotBlank(expectedCsvHeaders)) {
+            String[] expectedHeaders = expectedCsvHeaders.split(MappingConstants.COMA);
+            if (!isStringArraysEqual(expectedHeaders, actualCsvHeaders)) {
```

Comment:
Trivial comment but we try to  avoid !  in rd as those are not readable. Please use BooleanUtils